env-hooks/1.ros_package_path.bat.em

```diff
@@ -5,26 +5,39 @@ REM do not use EnableDelayedExpansion here, it messes with the != symbols
 setlocal disabledelayedexpansion
 echo from __future__ import print_function > _parent_package_path.py
 echo import os >> _parent_package_path.py
-echo env_name = 'CATKIN_WORKSPACES' >> _parent_package_path.py
-echo items = os.environ[env_name].split(';') if env_name in os.environ and os.environ[env_name] != '' else [] >> _parent_package_path.py
-echo path = '' >> _parent_package_path.py
-echo for item in items: >> _parent_package_path.py
-echo     path += ':' + (os.path.join(item, 'share') if item.find(':') == -1 else item.split(':')[1]) >> _parent_package_path.py
-echo print(path) >> _parent_package_path.py
+echo env_name = 'CMAKE_PREFIX_PATH' >> _parent_package_path.py
+echo paths = [path for path in os.environ[env_name].split(os.pathsep)] if env_name in os.environ and os.environ[env_name] != '' else [] >> _parent_package_path.py
+echo workspaces = [path for path in paths if os.path.exists(os.path.join(path, '.catkin'))] >> _parent_package_path.py
+echo # remove duplicate entries >> _parent_package_path.py
+echo unique_workspaces = [] >> _parent_package_path.py
```

Comment:
I would suggest to use a `set` here instead of a `list` since the order of elements doesn't matter and all items are unique.

Reply:
Thanks for the suggestion! updated to use `list(set())` + list comprehension to remove duplicate entries